include/exec/system_context.hpp

```diff
@@ -83,19 +100,118 @@ namespace exec {
     template <class _Sender>
     using __sender_data_t = decltype(stdexec::sync_wait(std::declval<_Sender>()).value());
 
+    /// A very simple spin lock. It is used in leu of a mutex because
+    /// instances of this class are constructed and never destroyed.
+    /// Doing that with a std::mutex might(?) leak system resources.
+    struct __spin_lock {
+      constexpr __spin_lock() noexcept = default;
+
+      void __lock() noexcept {
+        bool __is_locked = false;
+        // spin until we get the lock:
+        while (!__locked_.compare_exchange_strong(__is_locked, true)) {
+          std::this_thread::yield();
+          __is_locked = false;
+        }
```

Comment:
You probably want to use `atomic::wait()` and `atomic::notify_all()` to do the appropriate backoff here. Alternatively, you can use a binary_semaphore as a kind of spin-wait mutex.
Just initialize to have a count of 1 and then lock() calls acquire() and unlock() calls release().